sdk/core/core-lro/review/core-lro.api.md

```diff
@@ -9,36 +9,25 @@ import { AbortSignalLike } from '@azure/abort-controller';
 // @public (undocumented)
 export type CancelOnProgress = () => void;
 
-// @public (undocumented)
-export abstract class Poller<TState, TResult> implements PollerLike<TState, TResult> {
+// @public
+export abstract class Poller<TState extends PollOperationState<TResult>, TResult> implements PollerLike<TState, TResult> {
```

Comment:
We talked about this last week and decided it was a breaking change, I thought. This change would mean that anyone who had made their own Poller<> could be broken if it didn't follow the expected PollOperationState.

How do we know this new constraint is not a breaking change?

Reply:
This is a breaking change. What I tried to express was that the current use cases don't need to change considerably, since all of the states we have so far satisfy the PollOperationState constraint, but the pollers we have do need to implement getOperationState on their own to be able to use this.

Reply:
Yeah, this is breaking from the core-lro perspective (that is to say we should bump to 2.0 of core-lro and update the dependency from other libs) but I don't think any lib that consumes core-lro needs to major version bump because of how we've used it.